lib/AST/ASTContext.cpp

```diff
@@ -1641,20 +1642,25 @@ Optional<ModuleDependencies> ASTContext::getModuleDependencies(
     bool cacheOnly) {
   // Retrieve the dependencies for this module.
   if (cacheOnly) {
+    auto searchPathSet = getAllModuleSearchPathsSet();
     // Check whether we've cached this result.
     if (!isUnderlyingClangModule) {
       if (auto found = cache.findDependencies(moduleName,
-                                              ModuleDependenciesKind::SwiftTextual))
+                                              ModuleDependenciesKind::SwiftTextual,
+                                              searchPathSet))
         return found;
       if (auto found = cache.findDependencies(moduleName,
-                                              ModuleDependenciesKind::SwiftTextual))
+                                              ModuleDependenciesKind::SwiftTextual,
+                                              searchPathSet))
         return found;
       if (auto found = cache.findDependencies(moduleName,
-                                              ModuleDependenciesKind::SwiftPlaceholder))
+                                              ModuleDependenciesKind::SwiftPlaceholder,
+                                              searchPathSet))
         return found;
     }
     if (auto found = cache.findDependencies(moduleName,
-                                            ModuleDependenciesKind::Clang))
+                                            ModuleDependenciesKind::Clang,
+                                            searchPathSet))
```

Comment:
I'm not a big fan of duplicating the same function call for all enum cases. One technique we can use here is to define `ModuleDependenciesKind::First_Kind` and `ModuleDependenciesKind::Last_Kind` and use a for loop to iterate them. This applies to other duplications as well.

Reply:
Given that this PR does not introduce this behavior, can we pursue this refactoring outside the scope of this change? 

Even in this specific example there is additional complexity where simple iteration would not be enough (non-clang dependencies are skipped under a certain condition). And since this PR already re-structures quite a bit of code, I'd prefer to reduce further refactoring of existing patterns here.

Reply:
Sure, a following PR would be much appreciated!